server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java

```diff
@@ -129,6 +131,17 @@
             INDICES_RECOVERY_INTERNAL_LONG_ACTION_TIMEOUT_SETTING::get, TimeValue.timeValueSeconds(0),
             Property.Dynamic, Property.NodeScope);
 
+    /**
+     * recoveries would try to use files from available snapshots instead of sending them from the source node.
+     * defaults to `false`
+     */
+    public static final Setting<Boolean> INDICES_RECOVERY_USE_SNAPSHOTS_SETTING =
+        Setting.boolSetting("indices.recovery.use_snapshots", false, Property.Dynamic, Property.NodeScope);
```

Comment:
May as well default this to `true` since the repo-level setting defaults to `false`.